oidc-exchange.py

```diff
@@ -0,0 +1,156 @@
+import os
+from pathlib import Path
+import sys
+from textwrap import dedent
+from typing import NoReturn
+from urllib.parse import urlparse
+
+import id
+import requests
+
+_GITHUB_STEP_SUMMARY = Path(os.getenv("GITHUB_STEP_SUMMARY"))
+
+_TOKEN_RETRIEVAL_FAILED_MESSAGE = dedent(
+    """
+    OIDC token retrieval failed: {identity_error}
+
+    This generally indicates a workflow configuration error, such as insufficient
+    permissions. Make sure that your workflow has `id-token: write` configured
+    at either the workflow or job level, e.g.:
+
+    ```yaml
+    permissions:
+      id-token: write
+    ```
+    """
+)
+
+
+def die(msg: str) -> NoReturn:
+    with _GITHUB_STEP_SUMMARY.open("a") as io:
```

Comment:
oidc-exchange.py:30:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
```suggestion
    with _GITHUB_STEP_SUMMARY.open("a", encoding="utf-8") as io:
```